packages/react-native-jigsaw/src/components/Card.tsx

```diff
@@ -1,46 +1,232 @@
+          padding: ICON_CONTAINER_PADDING,
+          borderRadius: ICON_CONTAINER_SIZE,
+          backgroundColor: "rgba(0, 0, 0, 0.1)",
+        }}
+      >
+        <Pressable
+          disabled={!onPress}
+          onPress={onPress}
+          style={({ pressed }) => {
+            return [
+              {
+                opacity: pressed ? 0.8 : 1,
+              },
+            ];
+          }}
+        >
+          <IconOverride
+            name={icon}
+            size={ICON_SIZE}
+            color={theme.colors.surface}
+          />
+        </Pressable>
+      </Surface>
+    );
+  }
+);
 
 type Props = {
+  image?: string | ImageSourcePropType;
+  title?: string;
+  subtitle?: string;
+  description?: string;
+  textCentered: boolean;
+  icon?: string;
+  aspectRatio?: number;
+  elevation?: number;
   numColumns?: number;
-  children?: React.ReactNode;
-  onPress?: () => void;
+  theme: Theme;
+  titleStyle?: StyleProp<TextStyle>;
+  subtitleStyle?: StyleProp<TextStyle>;
+  descriptionStyle?: StyleProp<TextStyle>;
   style?: StyleProp<ViewStyle>;
-  theme: typeof theme;
-};
-
-const getWidth = (numColumns: number) => {
-  switch (numColumns) {
-    case 1:
-      return "33%";
-    case 2:
-      return "50%";
-    default:
-      return "100%";
-  }
+  onPress?: () => void;
+  onPressIcon?: () => void;
+  children?: React.ReactNode;
+  IconOverride?: typeof Icon;
 };
 
 const Card: React.FC<Props> = ({
-  numColumns = 3,
-  children,
-  onPress,
+  image = Config.cardImageUrl,
+  title,
+  subtitle,
+  description,
+  textCentered,
+  icon,
+  aspectRatio = 1.5,
+  elevation = 2,
   style,
-  ...rest
+  onPress,
+  onPressIcon,
+  titleStyle,
+  subtitleStyle,
+  descriptionStyle,
+  theme,
+  children,
+  IconOverride = null,
 }) => {
-  const width = getWidth(numColumns);
+  // Necessary to inject web-renderable Icons in buider.
+  const SelectedIcon = IconOverride || Icon;
+
+  const { backgroundColor: bgColor, padding, ...styles } = StyleSheet.flatten(
+    style || {}
+  );
+
+  const backgroundColor = bgColor ? bgColor : theme.colors.surface;
+  const innerPadding = padding ? padding : 12;
+
   return (
-    <Touchable
-      disabled={!onPress}
-      onPress={onPress}
-      style={[style, { width }]}
-      {...rest}
-    >
-      {children}
-    </Touchable>
+    <Surface style={[{ elevation, backgroundColor }, styles]}>
+      <Pressable
+        disabled={!onPress}
+        onPress={onPress}
+        style={({ pressed }) => {
+          return [
+            {
+              opacity: pressed ? 0.8 : 1,
+            },
+          ];
+        }}
+      >
+        <Image
+          style={{ aspectRatio }}
+          source={typeof image === "string" ? { uri: image } : image}
+          resizeMode="cover"
+        />
+        <View style={{ padding: innerPadding }}>
+          <View style={{ alignItems: textCentered ? "center" : "flex-start" }}>
+            {title || (title && title !== "") ? (
```

Comment:
Could this be simplified to this? I could be missing a case but I'm not sure when the right hand side of the `or` would hit
```suggestion
            {(title || title === "") ? (
```

Reply:
Typescript

Reply:
man typescript...